backend/src/middleware/authenticateJWT.js

```diff
@@ -11,15 +18,28 @@ const authenticateJWT = (req, res, next) => {
 
         jwt.verify(token, JWT_SECRET, (err, user) => {
             if (err) {
-                return res.sendStatus(403);  // Token invalid or expired
+                if (err.name === 'TokenExpiredError') {
+                    console.error('Error: Token has expired');
+                    return res.status(401).json({ message: 'Token expired. Please login again.' });
+                } else if (err.name === 'JsonWebTokenError') {
+                    console.error('Error: Invalid token');
+                    return res.status(401).json({ message: 'Invalid token. Authorization denied.' });
```

Comment:
Should this case result in a 403? Same question for `TokenExpiredError`.

Reply:
For this case its if the user was not able to be authenticated which i think it should be 401. If I am not wrong 403 is even if the user was able to authenticate.